data_model/derive/src/lib.rs

```diff
@@ -484,3 +485,45 @@ pub fn partially_tagged_deserialize_derive(input: TokenStream) -> TokenStream {
 pub fn has_origin_derive(input: TokenStream) -> TokenStream {
     has_origin::impl_has_origin(&parse_macro_input!(input))
 }
+
+/// Derive macro to implement `AssociatedConstant<T>` trait for enum variants where
+/// `T` is the type of discriminant.
+/// So perfectly this macro should be used together with `EnumDiscriminants` derive macro from `strum` crate.
+///
+/// It's the user responsibility to import `AssociatedConstant` trait.
+///
+/// # Attributes
+///
+/// `#[variant_discriminant(name(DiscriminantType))]` attribute is required.
+///
+/// # Examples
+///
+/// ```
+/// use iroha_data_model_derive::VariantDiscriminant;
+/// use iroha_data_model::AssociatedConstant;
+///
+/// #[derive(VariantDiscriminant)]
+/// #[variant_discriminant(name(MyEnumKind))]
+/// enum MyEnum {
+///     Unsigned(u32),
+///     String(String),
+///     Boolean(bool),
+/// }
+///
+/// #[derive(Debug, PartialEq, Eq)]
+/// enum MyEnumKind {
+///    Unsigned,
+///    String,
+///    Boolean,
+/// }
+///
+/// assert_eq!(<u32 as AssociatedConstant<MyEnumKind>>::VALUE, MyEnumKind::Unsigned);
+/// assert_eq!(<String as AssociatedConstant<MyEnumKind>>::VALUE, MyEnumKind::String);
+/// assert_eq!(<bool as AssociatedConstant<MyEnumKind>>::VALUE, MyEnumKind::Boolean);
+/// ```
```

Comment:
```suggestion
/// Derive macro implementing the [`AssociatedConstant<T>`](iroha_data_model::AssociatedConstrant) trait. 
/// # Notes
/// You should import `AssociatedConstant`. 
///
/// # Attributes
///
/// `#[variant_discriminant(name(DiscriminantType))]` attribute where `DiscriminantType` is the name of the `enum` that should serve as the discriminant enumeration. 
///
/// # Examples
///
/// ```
/// use iroha_data_model_derive::VariantDiscriminant;
/// use iroha_data_model::AssociatedConstant;
///
/// #[derive(VariantDiscriminant)]
/// #[variant_discriminant(name(MyEnumKind))]
/// enum MyEnum {
///     Unsigned(u32),
///     String(String),
///     Boolean(bool),
/// }
///
/// #[derive(Debug, PartialEq, Eq)]
/// enum MyEnumKind {
///    Unsigned,
///    String,
///    Boolean,
/// }
///
/// assert_eq!(<u32 as AssociatedConstant<MyEnumKind>>::VALUE, MyEnumKind::Unsigned);
/// assert_eq!(<String as AssociatedConstant<MyEnumKind>>::VALUE, MyEnumKind::String);
/// assert_eq!(<bool as AssociatedConstant<MyEnumKind>>::VALUE, MyEnumKind::Boolean);
/// ```
```